tests/unit_tests/simtel/test_simtel_io_event_writer.py

```diff
@@ -81,394 +101,376 @@ def validate_datasets(reduced_data, triggered_data, file_names, trigger_telescop
-        mock_trigger1,  # First event
-        mock_tracking1,
-        mock_tracking2,
-        mock_trigger2,  # Second event
-        mock_tracking3,
-    ]
+    assert len(lookup_table_generator.trigger_data) == initial_len
 
-    # Set initial shower info for triggered events
-    lookup_table_generator.shower = {"shower": 42}
-    lookup_table_generator.shower_id_offset = 1000
 
-    # Process the mock array event
-    lookup_table_generator._process_array_event(mock_array_event)
+def test_process_array_event_with_trigger_data(lookup_table_generator):
+    """Test processing array events and updating trigger data."""
+    # Setup test event
+    mock_array_event = create_array_event()
 
-    # Verify trigger information was processed correctly
-    assert len(lookup_table_generator.triggered_data.triggered_id) == 2
-    assert lookup_table_generator.triggered_data.triggered_id == [1042, 1042]  # 42 + 1000
+    # Add required shower data
+    lookup_table_generator.shower_data.append({"shower_id": 1, "event_id": 42, "file_id": 0})
+
+    # Process event
+    lookup_table_generator._process_array_event(mock_array_event, 0)
+
+    # Verify trigger data
+    assert len(lookup_table_generator.trigger_data) == 1
+    trigger_event = lookup_table_generator.trigger_data[0]
+    assert trigger_event["shower_id"] == 1
+    assert trigger_event["event_id"] == 42
+    assert trigger_event["telescope_list"] == one_two_three
 
-    # Verify telescope lists were stored correctly
-    assert len(lookup_table_generator.triggered_data.trigger_telescope_list_list) == 2
-    np.testing.assert_array_equal(
-        lookup_table_generator.triggered_data.trigger_telescope_list_list[0], np.array([1, 2])
-    )
-    np.testing.assert_array_equal(
-        lookup_table_generator.triggered_data.trigger_telescope_list_list[1], np.array([3, 4, 5])
-    )
 
-    # Verify tracking positions were processed correctly (using mean values)
-    assert len(lookup_table_generator.triggered_data.array_altitudes) == 1
-    assert len(lookup_table_generator.triggered_data.array_azimuths) == 1
-    np.testing.assert_almost_equal(
-        lookup_table_generator.triggered_data.array_altitudes[0],
-        0.4,  # mean of [0.3, 0.4, 0.5]
+def test_get_preliminary_nsb_level(lookup_table_generator):
+    """Test parsing NSB levels from filenames."""
+    assert lookup_table_generator._get_preliminary_nsb_level("dark_file.simtel.zst") == 1.0
+
+    assert lookup_table_generator._get_preliminary_nsb_level("half_nsb_file.simtel.zst") == 2.0
+
+    assert (
+        lookup_table_generator._get_preliminary_nsb_level("gamma_full_moon_file.simtel.zst") == 5.0
     )
-    # Test circular mean of azimuths
-    expected_azimuth1 = calculate_circular_mean([1.5, 1.7, 1.9])
-    np.testing.assert_almost_equal(
-        lookup_table_generator.triggered_data.array_azimuths[0], expected_azimuth1
+
+    assert lookup_table_generator._get_preliminary_nsb_level("file.simtel.zst") == 1.0
```

Comment:
Adding this comment here, but this will change soon anyway: Not sure it is good to assume a level dark if there is no indication at all in the filename.

Reply:
Yes, but until #1572 is implemented, this is the way we work. So I think we can keep it.

Reply:
Yes, this will be covered with the issue #1572 - for now this is our working assumption.